simbad/rotsearch/amore_search.py

```diff
@@ -217,50 +217,56 @@ def run(self, models_dir, nproc=2, shres=3.0, pklim=0.5, npic=50, rotastep=1.0,
             dat_models.append(info)
 
         sorted_dat_models = sorted(dat_models, key=lambda x: float(x.mw_diff), reverse=False)
+        n_files = len(sorted_dat_models)
+        chunk_size = simbad.rotsearch.get_chunk_size(n_files, chunk_size)
+        total_chunk_cycles = simbad.rotsearch.get_total_chunk_cycles(n_files, chunk_size)
 
         if submit_qtype == 'local':
             processes = nproc
         else:
             processes = submit_nproc
 
+        results = []
         iteration_range = range(0, n_files, chunk_size)
         for cycle, i in enumerate(iteration_range):
             logger.info("Working on chunk %d out of %d", cycle + 1,
                         total_chunk_cycles)
 
-            collector = ScriptCollector(None)
-            amore_files = []
-            with pool.Pool(processes=processes) as p:
-                [(collector.add(i[0]), amore_files.append(i[1])) for i in
-                 p.map(self, sorted_dat_models[i:i + chunk_size]) if i is not None]
-
-            results = []
-            if len(collector.scripts) > 0:
-                logger.info("Running AMORE tab/rot functions")
-                amore_logs, dat_models = zip(*amore_files)
-                simbad.util.submit_chunk(collector, self.script_log_dir, nproc, 'simbad_amore',
-                                              submit_qtype, submit_queue, True, monitor, self.rot_succeeded_log)
-
-                for dat_model, amore_log in zip(dat_models, amore_logs):
-                    base = os.path.basename(amore_log)
-                    pdb_code = base.replace("amore_", "").replace(".log", "")
-                    try:
-                        rotsearch_parser = simbad.parsers.rotsearch_parser.AmoreRotsearchParser(
-                            amore_log
-                        )
-                        score = simbad.core.amore_score.AmoreRotationScore(
-                            pdb_code, dat_model, rotsearch_parser.alpha, rotsearch_parser.beta, rotsearch_parser.gamma,
-                            rotsearch_parser.cc_f, rotsearch_parser.rf_f, rotsearch_parser.cc_i, rotsearch_parser.cc_p,
-                            rotsearch_parser.icp, rotsearch_parser.cc_f_z_score, rotsearch_parser.cc_p_z_score,
-                            rotsearch_parser.num_of_rot
-                        )
-                        if rotsearch_parser.cc_f_z_score:
-                            results += [score]
-                    except IOError:
-                        pass
-
+            if not self.solution:
```

Comment:
I'd change the logic here. If you wrote:
if self.solution:
    logger.info("Early termination criteria met, skipping chunk %d", cycle + 1)
    continue
Then it's clear what's going on and everything below doesn't need to be indented